### DISCUSSION TITLE
Multiple dialogs but not nested

### DISCUSSION THREAD

**Discussion by hyanmandian** [ariakit/ariakit#2703] [Questions] [answered]
In my app, I have a bunch of "global dialogs" that could be accessed on any part of my app even inside other dialogs. The problem here is that as soon as I open the first dialog and then open the second dialog (that is not nested in that case) everything is treated as an "inert" element and I cannot interact with any of those. I tried to use `getPersistentElements` for it, but I still need to only have one dialog interactable at a time.

So what is basically happening:

- Open the first dialog
- Everything else is inert
- Inside this first dialog, open another dialog (a global dialog)
- Everything is inert

Here is an example of it: https://stackblitz.com/edit/qp3fdq?file=dialog%2Findex.tsx

**Comment by diegohaz** [comment]
> So what is basically happening:
> 
> * Open the first dialog
> * Everything else is inert
> * Inside this first dialog, open another dialog (a global dialog)
> * Everything is inert

Hello! I'm not sure I understand what you want to achieve. Can you also outline the expected behavior using bullet points?

**Reply by hyanmandian** [reply]
Ok, sorry for the confusion, let me explain it:

https://github.com/ariakit/ariakit/assets/5044101/901b4055-7b0c-4f76-8e95-c6f136b479fb

Transcription: Let's say I have this global dialog that is some feature that the user could open in any part of our app and we have this local dialog for specific pages if we open each one individually it works as expected, but if we try to open both at the same time, so let's say we open this local dialog and after that, we open the global dialog, in that case, it sets "inert" in both elements and we can't interact with any of those

**Reply by diegohaz** [reply, marked as answer]
Got it! Thanks for the video.

That happens because the global dialog is already in the DOM when the local dialog opens. Ariakit will mark all existing elements in the body as external.

You can either render the global dialog dynamically or use `getPersistentElements` on the local dialog passing the global dialog: https://stackblitz.com/edit/qp3fdq-ewfuwe?file=dialog%2Findex.tsx%3AL21

**Reply by hyanmandian** [reply]
Awesome! Thanks for helping me @diegohaz . Option 1 works fine!
